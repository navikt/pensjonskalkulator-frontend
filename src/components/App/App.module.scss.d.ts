import globalClassNames from '../../style.d'
declare const classNames: typeof globalClassNames & {
  readonly main: 'main'
}
export = classNames
