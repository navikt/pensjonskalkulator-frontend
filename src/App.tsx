import { useState } from 'react'

import { Button } from '@navikt/ds-react'

import reactLogo from './assets/react.svg'

import styles from './App.module.scss'

function App() {
  const [count, setCount] = useState(0)

  return (
    <div className="App">
      <div>
        <a href="https://vitejs.dev" target="_blank">
          <img
            src="/pensjon/kalkulator/vite.svg"
            className={styles.logo}
            alt="Vite logo"
          />
        </a>
        <a href="https://reactjs.org" target="_blank">
          <img
            src={reactLogo}
            className={`${styles.logo} ${styles['logo--react']}`}
            alt="React logo"
          />
        </a>
      </div>
      <h1 className={styles.title}>Pensjonskalkulator</h1>
      <div className={styles.card}>
        <Button
          className={styles.button}
          onClick={() => setCount((c) => c + 1)}
        >
          count is {count}
        </Button>
        <p>Lorem ipsum dolor sit amet</p>
      </div>
    </div>
  )
}

export default App
