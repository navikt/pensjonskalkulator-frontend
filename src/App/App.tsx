import React from 'react'
import clsx from 'clsx'

import frameStyles from '../Frame/Frame.module.scss'
import { Pensjonsberegning } from '../Pensjonsberegning'

import styles from './App.module.scss'

export function App() {
  return (
    <main
      className={clsx(
        frameStyles.frame,
        frameStyles.frame_isFlex,
        frameStyles.frame_hasPadding,
        styles.main
      )}
    >
      <Pensjonsberegning/>
    </main>
  )
}
