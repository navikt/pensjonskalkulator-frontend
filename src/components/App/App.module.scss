@import '../../scss/variables';

.main {
  padding-top: var(--a-spacing-6);
  display: flex;
  align-items: start;
}

.headerGroup {
  padding: 0 var(--a-spacing-6);
}

.section {
  display: flex;
  flex-direction: column;
  gap: var(--a-spacing-4);
  padding: var(--a-spacing-6);
  text-align: start;
  margin-bottom: var(--a-spacing-6);

  @media (min-width: $a-breakpoint-sm) {
    padding: var(--a-spacing-10) var(--a-spacing-10);
  }
}
