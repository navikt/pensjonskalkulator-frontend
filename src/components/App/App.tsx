import React from 'react'

import { Heading } from '@navikt/ds-react'

import { TidligstMuligeUttak } from '@/components/TidligstMuligeUttak'

import styles from './App.module.scss'

export function App() {
  return (
    <main className={styles.main}>
      <div className={styles.headerGroup}>
        <Heading size="large" level="1" spacing>
          Pensjonskalkulator
        </Heading>
      </div>
      <TidligstMuligeUttak />
    </main>
  )
}
