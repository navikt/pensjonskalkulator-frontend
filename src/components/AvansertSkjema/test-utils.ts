export function getPreviousMonth(): string {
  const today = new Date()
  const currentMonth = parseInt(
    today.toLocaleDateString('en-CA').slice(5, 7),
    10
  ) // Extracts the month from the date string
  const previousMonth = currentMonth - 1
  return previousMonth.toString()
}
