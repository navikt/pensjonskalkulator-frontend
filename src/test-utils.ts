import renderer from 'react-test-renderer'

import { render } from '@testing-library/react'

export function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result as renderer.ReactTestRendererJSON
}

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {
    // wrap provider(s) here if needed
    wrapper: ({ children }) => children,
    ...options,
  })

export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
// override render export
export { customRender as render }
