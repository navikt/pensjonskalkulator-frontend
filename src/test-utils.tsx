import React, { PropsWithChildren } from 'react'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'

import { PreloadedState, createListenerMiddleware } from '@reduxjs/toolkit'
import { render, RenderOptions } from '@testing-library/react'

import { getTranslation_test } from '@/utils/__tests__/test-translations'

import { createUttaksalderListener } from './state/listeners/uttaksalderListener'
import {
  setupStore,
  RootState,
  AppStore,
  AppStartListening,
} from './state/store'
import { getTranslation_nb } from './translations/nb'

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
  hasRouter?: boolean
}

export const swallowErrors = (testFn: () => void) => {
  const error = console.error
  console.error = () => {}
  testFn()
  console.error = error
}

export const swallowErrorsAsync = async (testFn: () => Promise<void>) => {
  const cache = console.error
  console.error = () => {}
  await testFn()
  console.error = cache
}

function generateMockedTranslations() {
  const nbTranslations: Record<string, string> = getTranslation_nb()
  const testTranslations: Record<string, string> = getTranslation_test()
  const translationsInput = { ...nbTranslations, ...testTranslations }
  const translations: Record<string, string> = {}

  for (const key in translationsInput) {
    if (
      /<(?=.*? .*?\/ ?>|br|hr|input|!--|wbr)[a-z]+.*?>|<([a-z]+).*?<\/\1>|{[^}]+}/i.test(
        translationsInput[key]
      )
    ) {
      // for html or chunks keys: results in 'my_key' : 'my_key with some <html> or {chunk}'
      translations[key] = translationsInput[key]
    } else {
      // results in 'my_key' : 'my_key'
      translations[key] = key
    }
  }
  return translations
}

// Return an object with the store and all of RTL's query functions
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState, true),
    hasRouter = true,
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<unknown>): JSX.Element {
    return (
      <Provider store={store}>
        <IntlProvider locale="nb" messages={generateMockedTranslations()}>
          {hasRouter ? <MemoryRouter>{children}</MemoryRouter> : children}
        </IntlProvider>
      </Provider>
    )
  }

  const listenerMiddleware = createListenerMiddleware()
  createUttaksalderListener(
    listenerMiddleware.startListening as AppStartListening
  )

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'

export { renderWithProviders as render }
