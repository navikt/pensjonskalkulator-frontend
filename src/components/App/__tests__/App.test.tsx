import React from 'react'

import { waitFor } from '@testing-library/react'
import { describe, it } from 'vitest'

import { render, screen } from '../../../test-utils'
import { App } from '../App'

describe('App', () => {
  it('rendrer slik den skal, med main tag og Heading på riktig nivå', async () => {
    const result = render(<App />)

    await waitFor(() => {
      expect(screen.getByText('Hei Ola!')).toBeVisible()
      expect(result.asFragment()).toMatchSnapshot()
    })
  })
})
