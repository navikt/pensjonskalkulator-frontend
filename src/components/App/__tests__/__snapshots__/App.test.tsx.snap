// Vitest Snapshot v1

exports[`App > rendrer slik den skal, med main tag og Heading på riktig nivå 1`] = `
<DocumentFragment>
  <main
    class="_main_bbbd1d"
  >
    <div
      class="_headerGroup_bbbd1d"
    >
      <h1
        class="navds-heading navds-heading--large navds-typo--spacing"
      >
        forside.title
      </h1>
    </div>
    <svg
      aria-labelledby="loader-r0"
      class="navds-loader _loader__isCentered_d51639 navds-loader--3xlarge navds-loader--neutral"
      data-testid="loader"
      focusable="false"
      preserveAspectRatio="xMidYMid"
      viewBox="0 0 50 50"
    >
      <title
        id="loader-r0"
      >
        Henter tidligste mulige uttaksalder
      </title>
      <circle
        class="navds-loader__background"
        cx="25"
        cy="25"
        fill="none"
        r="20"
        xmlns="http://www.w3.org/2000/svg"
      />
      <circle
        class="navds-loader__foreground"
        cx="25"
        cy="25"
        fill="none"
        r="20"
        stroke-dasharray="50 155"
      />
    </svg>
  </main>
</DocumentFragment>
`;
